src/workerd/jsg/value.h

```diff
@@ -916,28 +916,40 @@ class ArrayBufferWrapper {
     // v8::ArrayBuffer::NewBackingStore() that accepts a deleter callback, and arrange for it to
     // delete an Array<byte> placed on the heap.
     //
-    // TODO(perf): We could avoid an allocation here, perhaps, by decomposing the kj::Array<byte>
-    //   into its component pointer and disposer, and then pass the disposer pointer as the
-    //   "deleter_data" for NewBackingStore. However, KJ doesn't give us any way to decompose an
-    //   Array<T> this way, and it might not want to, as this could make it impossible to support
-    //   unifying Array<T> and Vector<T> in the future (i.e. making all Array<T>s growable). So
-    //   it may be best to stick with allocating an Array<byte> on the heap after all...
     size_t size = value.size();
     if (size == 0) {
       // BackingStore doesn't call custom deleter if begin is null, which it often is for empty
       // arrays.
       return v8::ArrayBuffer::New(isolate, 0);
     }
     byte* begin = value.begin();
+    if (isolate->GetGroup().SandboxContains(begin)) {
```

Comment:
Perhaps this could be extracted into a separate utility since there are two places doing this same check and operation?